compiler/rustc_lint/src/builtin.rs

```diff
@@ -2853,8 +2859,8 @@ impl ClashingExternDeclarations {
                                     structurally_same_type_impl(
                                         seen_types,
                                         cx,
-                                        tcx.type_of(a_did),
-                                        tcx.type_of(b_did),
+                                        tcx.bound_type_of(a_did).subst_identity(),
+                                        tcx.bound_type_of(b_did).subst_identity(),
```

Comment:
it seems like there are possible `substs` here from the `(Adt(..), Adt(..))` above